src/util.c

```diff
@@ -15,6 +16,14 @@ void mg_random(void *buf, size_t len) {
     if (fread(buf, 1, len, fp) == len) done = true;
     fclose(fp);
   }
+#else
+  static uint64_t millis_last;
+  volatile uint64_t millis = mg_millis();
+  // Seed/reseed on timout
+  if (!millis_last || ((millis - millis_last) > MG_RESEED_PERIOD_MS)) {
+    srand((unsigned int)millis);
+    millis_last = millis;
+  }
 #endif
```

Comment:
My belief is that this will re-seed on call after a timeout.
So far we have two issues: the most important is getting a random ephemeral port on start, to avoid a known "flaw", the other is to generate a random MAC address on startup. That is, we just use it on startup.
What we need is a random seed, like e.g.: re-seeding on DHCP response would solve the ephemeral ports issue, but not the MAC issue.

Reply:
Correct, as discussed yesterday, even phy auto-negociation can be very predictable on some hardware and wont's solve it. This needs more investigation.

Reply:
Updated per discuss, reading heap/stack works, mixing timer makes it better.
That works on the NXP before network access, whereas mg_millis() was too predictable.
Thanks for the idea that's the draft implementation of it.